src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/BaseHeartbeatProperties.cs

```diff
@@ -0,0 +1,174 @@
+        /// restarted. 
+        /// 
+        /// <remarks>If a process is unstable and is being restared frequently, tracking this property
+        /// in the heartbeat would help to identify this unstability.
+        /// </remarks>
+        /// </summary>
+        private static Guid? uniqueProcessSessionId = null;
+
+        public string Name => "Base";
+
+        public bool IsKeyword(string keyword)
+        {
+            return this.DefaultFields.Contains(keyword, StringComparer.OrdinalIgnoreCase);
+        }
+
+        public Task<bool> SetDefaultPayload(IEnumerable<string> disabledFields, IHeartbeatProvider provider)
+        {
+            bool hasSetValues = false;
+            var enabledProperties = this.DefaultFields.Except(disabledFields);
+
+            foreach (string fieldName in enabledProperties)
+            {
+                // we don't need to report out any failure here, so keep this look within the Sdk Internal Operations as well
+                try
+                {
+                    switch (fieldName)
+                    {
+                        case "runtimeFramework":
+                            provider.AddHeartbeatProperty(fieldName, true, this.GetRuntimeFrameworkVer(), true);
+                            hasSetValues = true;
+                            break;
+                        case "baseSdkTargetFramework":
+                            provider.AddHeartbeatProperty(fieldName, true, this.GetBaseSdkTargetFramework(), true);
+                            hasSetValues = true;
+                            break;
+                        case "osVersion":
+                            provider.AddHeartbeatProperty(fieldName, true, this.GetRuntimeOsType(), true);
+                            hasSetValues = true;
+                            break;
+                        case "processSessionId":
+                            provider.AddHeartbeatProperty(fieldName, true, this.GetProcessSessionId(), true);
+                            hasSetValues = true;
+                            break;
+                        default:
+                            provider.AddHeartbeatProperty(fieldName, true, "UNDEFINED", true);
+                            break;
+                    }
+                }
+                catch (Exception ex)
+                {
+                    CoreEventSource.Log.FailedToObtainDefaultHeartbeatProperty(fieldName, ex.ToString());
+                }
+            }
+
+            return Task.FromResult(hasSetValues);
+        }
+
+        /// <summary>
+        /// This will return the current running .NET framework version, based on the version of the assembly that owns
+        /// the 'Object' type. The version number returned can be used to infer other things such as .NET Core / Standard.
+        /// </summary>
+        /// <returns>a string representing the version of the current .NET framework</returns>
+        private string GetRuntimeFrameworkVer()
```

Comment:
I forgot what was the conclusion of our prior discussions on this topic, apologies if this is repeat!.
in my asp.net core app, targetting netcoreapp2.0, System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription returns the following
_".NET Core 4.6.26020.03"_

If there a way to return NETCORE2.0 ?

Also imp: 
Please provide example values for diff cases as a comment here.

Reply:
Let's take this field out altogether until we find a concrete method that we can map from runtime version to NETCORE / NETSTANDARD / NETFRAMEWORK versions.